### PR TITLE
token refresh example

### PR DESCRIPTION
**Pull request by cadebward**
Hey there! I took out the minimum amount of code from our token refresh implementation and tried to come up with a straight-forward example. Maybe after a bit of back and forth, when you're happy with the code in `client.ex`, I can proceed to write up the tutorial and tests to go with it.

Let me know what you think!

### CODE REVIEW

File: examples/token_refresh/client.ex
Comment (the-mikedavis) [inline examples/token_refresh/client.ex:17]: It might make sense to wrap `connect/2` in a helper that makes the token, so something like:

```elixir
defp connect_with_token(socket) do
  new_token = make_new_token()

  socket =
    update(socket, :config, fn config ->
      uri =
        config
        |> Keyword.get(:uri)
        |> URI.parse()
        |> Map.put(:query, "token=#{new_token}")
        |> URI.to_string()
  
      Keyword.put(config, :uri, uri)
    end)

  connect(socket, socket.assigns.config)
end
```

So the connection would look like

```elixir
new_socket()
|> assign(:config, config)
|> connect_with_token()
```

What do you think?

Comment (cadebward) [reply examples/token_refresh/client.ex]: Love it! That feels much cleaner. I've pushed up those changes and added a `README`. Let me know what you think. Once everything has your stamp of approval, I'll add tests! 

Thanks!

Comment (the-mikedavis) [reply examples/token_refresh/client.ex]: Nice, this looks great!

**Comment by cadebward** [comment]
I've added a pretty basic test. Any ideas of how to test it further? 🤷

**Comment by the-mikedavis** [comment]
I was fiddling with adding an `assert_connect` helper to `Slipstream.SocketTest` (https://github.com/NFIBrokerage/slipstream/commit/9e0c360b3a79f3308ee9a926f2cf5024b31b13ca) for testing reconnect behavior. I think that would help out here but I'm not sure yet how it should interact with the similar `Slipstream.SocketTest.accept_connect/1` (maybe they should be consolidated into one helper?). I'll swing back around to this example if that ends up making its way into the API.

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
Thanks for working on this @cadebward!

**Pull request merged**
